### PR TITLE
ALSA: hda: documentation, cleanups and fixes for the hdac_ext stream allocation/release

### PR DESCRIPTION
**Pull request by plbossart**
While trying to figure out how this hdac_stream is supposed to work, I found unused symbols and documented my understanding of this library/transitions.

The last part of this PR adds fixes to simplify the logic and handle the coupled mode.

### CODE REVIEW

File: sound/hda/hdac_stream.c [hunk 1]
```diff
@@ -13,6 +13,39 @@
...
+ * the hdac_stream library is intended to be used with the following
+ * transitions. The states are not formally defined in the code but loosely
+ * inspired by boolean variables. Note that the 'prepared' field is not used
+ * in this library but by the callers during the hw_params/prepare transitions
+ *
+ *			   |
+ *	stream_init()	   |
+ *			   v
+ *			+--+-------+
+ *			|   free   |
```
Comment (ranj063) [inline sound/hda/hdac_stream.c]: do we have a free flag @plbossart this is just !opened isnt it?

Comment (plbossart) [reply sound/hda/hdac_stream.c]: that's right. I renamed that state 'unused'.

**Comment by plbossart** [comment]
@crojewsk feeback welcome, and even better if you can take that PR for a spin. We don't have the ability to test this, only the Skylake and AVS driver make use of this hdac_ext_stream library for now.

**Comment by crojewsk** [comment]
I'll need to get used to to these mentions : ) Will schedule a test run tomorrow.

**Comment by plbossart** [comment]
replaced by https://github.com/thesofproject/linux/pull/3805, with the mixed allocation removed for now.

**Pull request closed without merging**